js/actions/index.ts

```diff
@@ -75,7 +75,8 @@ export function fetchAndObserveData() {
   const runKeyValue = urlParam("runKey");
   if (runKeyValue) {
     const activity = urlParam("activity") || "";
-    const source = activity ? makeSourceKey(activity) : "";
+    const sourceKey = urlParam("sourceKey") || "";
+    const source = sourceKey ? sourceKey : (activity ? makeSourceKey(activity) : "");
```

Comment:
Can you refactor the function https://github.com/concord-consortium/portal-report/blob/5355a12ee0f21b8b2fddbe58d56be00b6e5f343d/js/api.ts#L79-L83
So it takes a a url like `getSourceKey(url: string)`
Then you can use that function here and in the two other places where `getSourceKeyFromOffering` was used before.